manifests/init.pp

```diff
@@ -264,6 +265,16 @@
     $apache_https_key = undef
     $apache_https_ca = undef
     $apache_https_chain = undef
+
+    if $foreman_proxy::registration_url {
+      $client_facing_servername = foreman_proxy_content::host_from_url($foreman_proxy::registration_url)
+
+      unless $client_facing_servername in $certs::cname {
```

Comment:
Can we add `$serveraliases = $certs::apache::cname` around line 260 and an equivalent around line 247 (there you can probably use `undef` since it doesn't matter, or `$foreman::config::apache::serveraliases` if you want to have the proper value).

Then here you can use:
```suggestion
      unless $client_facing_servername in $serveraliases {
```

I was hoping to pass `serveraliases` to the `pulpcore` class around line 295, but see I'd need to release puppet-pulpcore for that. Still, I wouldn't mind doing that so we have a correct setup where Apache is aware of the server names it should accept.

Reply:
Sure thing! I've added that specification

Reply:
Looks like I opened https://github.com/theforeman/puppet-pulpcore/pull/272 in early November, but it was never reviewed. Just updated now. Please make sure you set that as a lower bound in metadata.json as well.

Reply:
With my current diff on this PR, I am not seeing how it would put a requirement on https://github.com/theforeman/puppet-pulpcore/pull/272 ? That feels like a separate change.

Reply:
That's this part that I mentioned:
> I was hoping to pass serveraliases to the pulpcore class around line 295, but see I'd need to release puppet-pulpcore for that. Still, I wouldn't mind doing that so we have a correct setup where Apache is aware of the server names it should accept.

You're right you didn't implement that, but I think we should.

Reply:
I added this change as a additional commit, while it requires the previous work it's not required to tackle the original commit. The second commit contains the `metadata.json` update as well.